rust/feature-flags/src/cohorts/cohort_operations.rs

```diff
@@ -243,8 +243,16 @@ fn evaluate_cohort_values(
         "AND" | "property" => {
             for filter in &values.values {
                 if filter.is_cohort() {
-                    // Handle cohort membership check
-                    if !apply_cohort_membership_logic(&[filter.clone()], cohort_matches)? {
+                    // Handle cohort membership check with negation
+                    let cohort_result =
+                        apply_cohort_membership_logic(&[filter.clone()], cohort_matches)?;
+                    // Apply negation if specified
+                    let final_result = if filter.negation.unwrap_or(false) {
+                        !cohort_result
+                    } else {
+                        cohort_result
+                    };
+                    if !final_result {
                         return Ok(false);
                     }
```

Comment:
nit: This could be simplified to:

```suggestion
                    if cohort_result == filter.negation.unwrap_or(false) {
                        return Ok(false);
                    }
```